sqlmesh/utils/dag.py

```diff
@@ -30,7 +31,8 @@ def add(self, node: T, dependencies: t.Optional[t.Iterable[T]] = None) -> None:
             node: The node to add.
             dependencies: Optional dependencies to add to the node.
         """
-        self._sorted = None
+        self._sorted = []
```

Comment:
IMHO I preferred the explicit `None` way. This way we have a clear delineation between an empty DAG and missing cache.